src/main/java/ui/MapEditorToolController.java

```diff
@@ -0,0 +1,653 @@
+            addingNeighbor = false;
+            removingNeighbor = false;
+            makingNew = false;
+        }
+    }
+
+    @FXML
+    /**
+     * update a node's X coordinate, both visually and in the node's properties
+     */
+    void updateNodeX(ActionEvent event)
+    {
+        try
+        {
+            currentButton.setLayoutX(Double.parseDouble(xField.getText()));
+            currentNode.setX(Double.parseDouble(xField.getText()));
+            //track that this node has been modified
+            if(!modifiedNodesList.contains(currentNode)){
+                modifiedNodesList.add(currentNode);
+            }
+        }
+        catch (NumberFormatException e)
+        {
+            //TODO: need more exception handling?
+            System.out.println("Not a double");
+        }
+    }
+
+    @FXML
+    /**
+     * update a node's Y coordinate, both visually and in the node's properties
+     */
+    void updateNodeY(ActionEvent event) {
+        try
+        {
+            if(currentButton != null && currentNode != null)
+            {
+                currentButton.setLayoutY(Double.parseDouble(yField.getText()));
+                currentNode.setY(Double.parseDouble(yField.getText()));
+                //track that this node has been modified
+                if(!modifiedNodesList.contains(currentNode)){
+                    modifiedNodesList.add(currentNode);
+                }
+            }
+        } catch (NumberFormatException e){
+            //TODO: need more exception handling?
+            System.out.println("Not a double");
+        }
+    }
+
+    @FXML
+    /**
+     * update a node's Name string
+     */
+    void updateNodeData(ActionEvent event)
+    {
+        try
+        {
+            if(currentButton != null && currentNode != null)
+            {
+                ArrayList<String> data = new ArrayList<String>();
+                data.add(nameField.getText());
+                data.add(typeField.getText());
+                currentNode.setData(data);
+                //track that this node has been modified
+                if(!modifiedNodesList.contains(currentNode))
+                {
+                    modifiedNodesList.add(currentNode);
+                }
+            }
+        }
+        catch (NumberFormatException e)
+        {
+            //TODO: need more exception handling?
+            System.out.println("Not a double");
+        }
+    }
+
+
+    @FXML
+    /**
+     * Push newly created nodes into the databasecontroller's node table.
+     *
+     * This function will:
+     *  - insert all newly created nodes into the database
+     *  - update all modified nodes that were loaded in from the database
+     *  - insert all new neighbor relations that have been created
+     *  - delete all neighbor relations that have been deleted
+     * TODO: nodes loaded in from the table and modified should also be pushed.
+     */
+    void pushChangesToDatabase(ActionEvent event)
```

Comment:
On the UI, the button should be named "Save" - the name is much more user-friendly